src/device/usbd.c

```diff
@@ -346,12 +350,59 @@ static bool process_control_request(uint8_t rhport, tusb_control_request_t const
 {
   usbd_control_set_complete_callback(NULL);
 
+  // First pre-compute which driver is in charge of the target (and do bounds checking)
+
+  // FIXME: Should itf and ep be 16-bit since the request's value and index are 16-bit?
+  // If so, these need byte swapping on big-endian platforms.
+  uint16_t itf = 0xFFu;

```

Comment:
This code will change the behavior in some cases. It prevents the vendor class from handling a request where the recipient is EP or ITF, and the wIndex is invalid.

Reply:
please don't change it, the vendor request will always be forwarded into vendor function. wValue, wIndex has different meaning depending on the request type. In case of Vendor, they can be just about anything (not limited to either ep or itf).

Reply:
I see.... I have not yet discovered a use case of vendor requests. In USBTMC, I'm ignoring them for now (they have their own vendor specification).